zeppelin-web-vue/src/services/api.js

```diff
@@ -0,0 +1,33 @@
+export default {
+  getWebsocketUrl () {
+    let wsProtocol = location.protocol === 'https:' ? 'wss:' : 'ws:'
+    return wsProtocol + '//' + location.hostname + ':' + this.getWSPort() + '/ws'
+  },
+
+  getWSPort () {
+    if (location.hostname.includes('qubole')) {
+      if (location.protocol === 'https:') {
+        return 443
+      }
+    }
+    return this.getPort()
+  },
+
+  getPort () {
+    let port = Number(location.port - 1)
+    if (!port) {
+      port = 80
+      if (location.protocol === 'https:') {
+        port = 443
+      }
+    }
+    if (port === 3333 || port === 9000) {
```

Comment:
Where does `3333` and `9000` come from ?

Reply:
I have refactored the logic. Now one can update the ports through .env file.